Source/WebCore/platform/gstreamer/WebKitFliteSourceGStreamer.cpp

```diff
@@ -0,0 +1,263 @@
+/* There is no header for Flite functions below */
+void usenglish_init(cst_voice*);
+cst_lexicon* cmulex_init(void);
+cst_voice* register_cmu_us_kal(const char*);
+cst_voice* register_cmu_us_awb(const char*);
+cst_voice* register_cmu_us_rms(const char*);
+cst_voice* register_cmu_us_slt(const char*);
+}
+
+using namespace WebCore;
+
+typedef struct _WebKitFliteSrcClass WebKitFliteSrcClass;
+typedef struct _WebKitFliteSrcPrivate WebKitFliteSrcPrivate;
+
+#define WEBKIT_FLITE_SRC_CLASS(klass) (G_TYPE_CHECK_CLASS_CAST((klass), WEBKIT_TYPE_FLITE_SRC, WebKitFliteSrcClass))
+#define WEBKIT_IS_FLITE_SRC(obj) (G_TYPE_CHECK_INSTANCE_TYPE((obj), WEBKIT_TYPE_FLITE_SRC))
+#define WEBKIT_IS_FLITE_SRC_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE((klass), WEBKIT_TYPE_FLITE_SRC))
+
+struct _WebKitFliteSrcClass {
+    GstBaseSrcClass parentClass;
+};
+
+struct _WebKitFliteSrc {
+    GstBaseSrc parent;
+    WebKitFliteSrcPrivate* priv;
+};
+
+struct _WebKitFliteSrcPrivate {
+    struct StreamingMembers {
+        bool didLoadUtterance;
+        GRefPtr<GstAdapter> adapter;
+    };
+    DataMutex<StreamingMembers> dataMutex;
+
+    GstAudioInfo info;
+    String text;
+    cst_voice* voice;
+};
+
+GType webkit_flite_src_get_type(void);
+
+GST_DEBUG_CATEGORY_STATIC(webkit_flite_src_debug);
+#define GST_CAT_DEFAULT webkit_flite_src_debug
+
+#define DEFAULT_SAMPLES_PER_BUFFER 1024
+
+static GstStaticPadTemplate srcTemplate = GST_STATIC_PAD_TEMPLATE("src",
+    GST_PAD_SRC,
+    GST_PAD_ALWAYS,
+    GST_STATIC_CAPS("audio/x-raw, "
+        "format = (string) " GST_AUDIO_NE(S16) ", "
+        "layout = (string) interleaved, "
+        "rate = (int) 48000, " "channels = (int) [1, 8]")
+);
+
+#define webkit_flite_src_parent_class parent_class
+WEBKIT_DEFINE_TYPE_WITH_CODE(WebKitFliteSrc, webkit_flite_src, GST_TYPE_BASE_SRC,
+    GST_DEBUG_CATEGORY_INIT(webkit_flite_src_debug, "webkitflitesrc", 0, "flitesrc element"));
+
+static cst_voice* defaultVoice()
+{
+    // The first voice on the registered voice list is the default.
+    return flite_voice_list ? val_voice(val_car(flite_voice_list)) : nullptr;
+}
+
+static void webkitFliteSrcReset(WebKitFliteSrc* src)
+{
+    WebKitFliteSrcPrivate* priv = src->priv;
+
+    DataMutexLocker members { priv->dataMutex };
+    gst_adapter_clear(members->adapter.get());
+    members->didLoadUtterance = false;
+}
+
+static void webkitFliteSrcConstructed(GObject* object)
+{
+    WebKitFliteSrc* src = WEBKIT_FLITE_SRC(object);
+    WebKitFliteSrcPrivate* priv = src->priv;
+
+    /* we operate in time */
+    gst_base_src_set_format(GST_BASE_SRC(src), GST_FORMAT_TIME);
+    gst_base_src_set_blocksize(GST_BASE_SRC(src), -1);
+    gst_base_src_set_automatic_eos(GST_BASE_SRC(src), FALSE);
+
+    DataMutexLocker members { priv->dataMutex };
+    members->adapter = adoptGRef(gst_adapter_new());
+
+    ensureFliteVoicesInitialized();
+}
+
+static GstStateChangeReturn webkitFliteSrcChangeState(GstElement* element, GstStateChange transition)
```

Comment:
Is this worth implementing only for logging purpose?

Reply:
Removed.